server/socket.js

```diff
@@ -0,0 +1,78 @@
+const { logger } = require('./logger');
+const EventEmitter = require('events');
+const jwt = require('jsonwebtoken');
+const env = require('./env.js');
+
+class Socket extends EventEmitter {
+    constructor(socket) {
+        super();
+
+        this.socket = socket;
+        this.user = socket.request.user;
+
+        socket.on('error', this.onError.bind(this));
+        socket.on('authenticate', this.onAuthenticate.bind(this));
+        socket.on('disconnect', this.onDisconnect.bind(this));
+    }
+
+    get id() {
+        return this.socket.id;
+    }
+
+    // Commands
+    registerEvent(event, callback) {
+        this.socket.on(event, this.onSocketEvent.bind(this, callback));
+    }
+
+    joinChannel(channelName) {
```

Comment:
This whole file is some sort of socket wrapper they were using for handling socket.io. I'm not entirely sure of its purpose and opted to just use socket.io directly for now. This may make more sense as we look more closely at communication between gamenodes.